### PR TITLE
Improve README to compile JICMP

### PR DESCRIPTION
**Pull request by indigo423**
Create README which explains how to checkout git submodules and what is required to compile JICMP. Give some hints about installing the compiled library as non-root user. Migrate from Markdown to AsciiDoc.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+[source]
+----
+cd jicmp
+----
+
+.Initialize and update the git submodule.
+[source]
+----
+git submodule init
+git submodule update
```
Comment (fooker) [inline README.adoc]: `git submodule update --init --recursive` does this in one line

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+.Update generated configuration files with
+[source]
+----
+autoreconf -fvi
+----
+
+.Generate make files and compile JICMP
+[source]
+----
+./configure
```
Comment (fooker) [inline README.adoc]: We should install to `/usr` using `./configure--prefix=/usr` as some distros don't include `usr/local` to the library search path by default.

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+_Mac OS X_ supports non-root _ICMP_ through the _SOCK\_DRGAM_ interface, which _JICMP_ uses by default.
+
+_Linux_ supports this as well, but you additionally need to set a sysctl _OID_ to allow ping for non-root users.
+
+You can set this temporarily by running: 
+ 
+[source]
+----
+sysctl -w net.ipv4.ping_group_range="0 429496729"
```
Comment (fooker) [inline README.adoc]: If we recommend this, we should limit to the system accounts (0 - 1024).

File: README.adoc (hunk 1 above)
Comment (indigo423) [inline README.adoc]: thank you, fixed also in the JICMP6

File: README.adoc (hunk 2 above)
Comment (indigo423) [inline README.adoc]: Make sense to me, should I add a hint, if prefix is not given it will use /usr/local?

File: README.adoc (hunk 3 above)
Comment (fooker) [inline README.adoc]: We should link to the documentation: https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt

File: README.adoc (hunk 2 above)
Comment (fooker) [inline README.adoc]: AFAIK its default to autoconf.

File: README.adoc (hunk 3 above)
Comment (indigo423) [inline README.adoc]: I'll fix this as well and adapt to JICMP6 it is basically exactly the same. Thanks for good input.

Comment (indigo423) [inline README.adoc]: The link to ip-sysctl.txt does not contain the description for `ping_group_range`. All I've found was a patch like this: https://lkml.org/lkml/2011/5/18/305. Any ideas why this is not in the latest docs?

Comment (fooker) [inline README.adoc]: You are right. Looks like the patch was not merged and it's not include in upstream kernels.

Comment (indigo423) [inline README.adoc]: Should keep the range for whole world then?

File: README.adoc (hunk 2 above)
Comment (indigo423) [inline README.adoc]: Fixed

Comment (RangerRick) [inline README.adoc]: Oop, didn't see you already have comments about this.  Installing hand-built files into the system library directory is bad practice.  If users want to do it themselves that's their problem but it should definitely not be the default suggestion.  Better to document how to make sure OpenNMS `install -dis` knows how to search in `/usr/local/lib` (`install -l /usr/local/lib -dis`).

**Comment by pbrane** [comment]
> On Sep 27, 2016, at 10:37 AM, Benjamin Reed <notifications@github.com> wrote:
> 
> Please never recommend users install hand-compiled files into /usr/

I thought that seemed strange, too.

**Comment by fooker** [comment]
Then we should document how to bring this to the search path (or link to the docs)

**Pull request merged**